lib/rules/no-lonely-if.js

```diff
@@ -52,9 +51,26 @@ module.exports = {
                                 return null;
                             }
 
+                            if (
+                                node.consequent.type !== "BlockStatement" && lastIfToken.value !== ";" && tokenAfterElseBlock &&
+                                (
+                                    node.consequent.loc.end.line === tokenAfterElseBlock.loc.start.line ||
+                                    /^[(\[+-`]/.test(tokenAfterElseBlock.value) ||
+                                    lastIfToken.value === "++" ||
+                                    lastIfToken.value === "--"
+                                )
+                            ) {
```

Comment:
I agree. What would you recommend that the function do? It seems like many of the ASI checks are checking slightly different things, so I wasn't sure about the best way to generalize the behavior.